From: SeongJae Park <sj@kernel.org>
Date: Sat, 10 May 2025 14:58:09 -0700
Subject: [PATCH] ==== damon_stat ====

mm/damon: introduce a DAMON module for access monitoring that just works

Access monitoring requires manual controls.  Introduce a static kernel
module for making it just works.  The module can be enabled without
manual setup and provides practical but simple and intuitive access
pattern metrics, namely memory idle time percentiles and estimatd memory
bandwidth.

Background
----------

DAMON was initially developed for access monitoring, and later extended
for access-aware system operations.  Ironically, it is being used for
access-aware system operations more than access monitoring.  Since DAMON
follows the random evolution method, it

Solution
--------

Discussions
-----------

There is no way to simply monitor data access pattern

DAMON provides monitoring results that fine-grained for memory
management.  For basic observability, though, it is too much
information.  The control is also arguably too complicated.  Introduce a
static kernel module that can easily be enabled and used.

The module makes its usage simple by providing parameter for turning it
on, which can be set at boot time.  The module uses monitoring intervals
auto-tuning to make it just works while consuming only minimum sytem
resource.

The module defines two simple access pattern metrics, namely page
idletime percentiles and estimated memory bandwidth.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_3m8mg2xr | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_3m8mg2xr

diff --git a/hkml_cv_bogus/hkml_cv_bogus_3m8mg2xr b/hkml_cv_bogus/hkml_cv_bogus_3m8mg2xr
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

